src/win32.c

```diff
@@ -1032,4 +1032,31 @@ gchar *win32_get_user_config_dir(void)
 	return g_build_filename(g_get_user_config_dir(), "geany", NULL);
 }
 
+
+void win32_make_argc_and_argv_in_utf8(gint *pargc, gchar ***pargv)
```

Comment:
Actually recent GLib (2.40+) have `g_win32_get_command_line()` which basically does just that.

(and doesn't bother with checking for error in the conversion, so it might be "alright" -- or maybe they find it OK because they don't return arg count and expect callers to use `g_strv_length()` which that would simply truncate the list, not sure)